common/repositories/inventory_snapshots_repository.h

```diff
@@ -46,16 +47,42 @@ class InventorySnapshotsRepository: public BaseInventorySnapshotsRepository {
 	// Custom extended repository methods here
 	static int64 CountInventorySnapshots(Database& db)
 	{
-		const std::string& query = "SELECT COUNT(*) FROM (SELECT * FROM `inventory_snapshots` a GROUP BY `charid`, `time_index`) b";
+		const std::string &query =
+			"SELECT COUNT(*) FROM (SELECT * FROM `inventory_snapshots` a GROUP BY `character_id`, `time_index`) b";
 
 		auto results = db.QueryDatabase(query);
-
 		if (!results.Success() || !results.RowCount()) {
 			return -1;
 		}
 
-		auto row = results.begin();
+		auto        row   = results.begin();
+		const int64 count = Strings::ToBigInt(row[0]);
+
+		if (count > std::numeric_limits<int>::max()) {
+			return -2;
+		}
+
+		if (count < 0) {
+			return -3;
```

Comment:
Magic number ?

Reply:
It would appear so.  I moved the zonedb functions into the repo only, and did not change the detail.